modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowDefinitionScanner.java

```diff
@@ -124,6 +154,23 @@ public boolean accept(File arg0, String name) {
       bundleCtx.registerService(ReadinessIndicator.class.getName(), new ReadinessIndicator(), properties);
       isWFSinitiliazed = true;
     }
+
+    // send update message
+    // Felix file install thread doesn't have user/organization, but this is required to send the message (though we
+    // don't actually need it in this particular use case). So we have to create a dummy user/organization...
+    final Organization dummyOrg = new JpaOrganization();
+    SecurityUtil.runAs(securityService, dummyOrg, SecurityUtil.createAnonymousUser(dummyOrg), () ->
+      messageSender.sendObjectMessage(
+        WorkflowDefinitionItem.WORKFLOW_DEFINITION_QUEUE,
+        MessageSender.DestinationType.Queue,
+        new WorkflowDefinitionItem(def.getId(),
+            def.getStateMappings()
+                .stream()
+                .map(m -> new SerializableWorkflowStateMapping(m.getState().name(), m.getValue()))
+                .toArray(SerializableWorkflowStateMapping[]::new)
+        )
+      )
+    );
```

Comment:
Do we really need to send messages for this? I feel like it would be nicer to keep and update the workflowStateMappings somewhere in the WorkflowDefinitionScanner, create an interface for the Scanner in workflow.api (as the IndexService and others already depends on that module for the WorkflowService) and then ask the scanner to do the mapping or hand the mappings over each time? I feel like Event shouldn't hold that much knowledge (although it already does for other things...).

Reply:
(Also does this mean that on system start, there's a message sent for each workflow definition?)